packages/shared/src/components/widgets/PostUsersHighlights.tsx

```diff
@@ -106,18 +103,19 @@ const Image = (props: ImageProps) => {
 };
 
 export const UserHighlight = (props: UserHighlightProps): ReactElement => {
+  const { userType, ...source } = props;
   const {
     id,
-    handle,
     name,
-    username,
     permalink,
-    userType = UserType.Source,
-    reputation,
     allowSubscribe = true,
     className,
     showReputation = false,
-  } = props;
+  } = source;
+
+  const typeHandle = 'handle' in source ? source.handle : source.username;
+  const reputation = 'reputation' in source ? source.reputation : NaN;
```

Comment:
Those props do not exist in every prop type so need for initial check before initialize